### ISSUE TITLE
Smidge bundles aren't always invalidated between backoffice versions

### ISSUE THREAD

**Issue by p-m-j** [umbraco/Umbraco-CMS#11698]
### Which *exact* Umbraco version are you using? For example: 9.0.1 - don't just write v9

> don't just write v9

v9

### Bug summary

See title, we're only regenerating bundles OOTB when runtime level is RuntimeLevel.Upgrade (i.e. migrations required) and apararently even that isn't working as expected.

### Steps to reproduce

1. Create umbraco 9.0.1 site with dotnet new templates
1. Remove Umbraco:Cms:RuntimeMinification section from appsettings.Development.json (so it runs like a released site would by default).
1. Add component from snippet below.
1. dotnet run site & Complete install process
    1.  should see "CacheBuster: 1" in console can also see request for umbraco.controllers.min.js?d=1 in network tab for backoffice
1. Update package ref in csproj to `<PackageReference Include="Umbraco.Cms" Version="9.1.0" />`
1. dotnet run site
    1.  should see "CacheBuster: 1" in console can also see request for umbraco.controllers.min.js?d=1 in network tab for backoffice
1. Upgrade is required - run upgrade
    1.  should see "CacheBuster: 1" in console can also see request for umbraco.controllers.min.js?d=1 in network tab for backoffice
1. Restart site 
    1.  should see "CacheBuster: 1" in console can also see request for umbraco.controllers.min.js?d=1 in network tab for backoffice 
    2. wat? the current setup is supposed to use config manipulator to nuke version on upgrade (only when there are migrations to run). This only occurs *if there is already a property there to replace.* If appsettings.json doesn't contain "Umbraco:CMS:RuntimeMinification:Version" then config manipulator will not write on upgrade.

```csharp
public class DemoComposer : ComponentComposer<DemoComponent> { }

public class DemoComponent : IComponent
{
    private readonly ILogger<DemoComponent> _log;
    private readonly IRuntimeMinifier _runtimeMinifier;

    public DemoComponent(ILogger<DemoComponent> log, IRuntimeMinifier runtimeMinifier)
    {
        _log = log;
        _runtimeMinifier = runtimeMinifier;
    }

    public void Initialize()
    {
        _log.LogWarning("CacheBuster: {v}",_runtimeMinifier.CacheBuster);
    }

    public void Terminate()
    {
    }
}
```

### Expected result / actual result

We should invalidate cached js bundles when they change e.g. pretty much every single release.

**Comment by p-m-j** [comment]
Note 9.0.1 -> 9.1.0 does have a migration and cache buster does update but only after completing upgrade and restarting.
We want to invalidate in production without a restart.

**Comment by nikolajlauridsen** [comment]
Fixed in #11700 🎉
